core-primitives/test/src/mock/onchain_mock.rs

```diff
@@ -84,6 +97,62 @@ impl GetStorageVerified for OnchainMock {
 	}
 }
 
+impl EnclaveAttestationOCallApi for OnchainMock {
+	fn sgx_init_quote(&self) -> SgxResult<(sgx_target_info_t, sgx_epid_group_id_t)> {
+		todo!()
+	}
+
+	fn get_ias_socket(&self) -> SgxResult<i32> {
+		Ok(42)
+	}
+
+	fn get_quote(
+		&self,
+		_sig_rl: Vec<u8>,
+		_report: sgx_report_t,
+		_sign_type: sgx_quote_sign_type_t,
+		_spid: sgx_spid_t,
+		_quote_nonce: sgx_quote_nonce_t,
+	) -> SgxResult<(sgx_report_t, Vec<u8>)> {
+		todo!()
+	}
+
+	fn get_update_info(
+		&self,
+		_platform_info: sgx_platform_info_t,
+		_enclave_trusted: i32,
+	) -> SgxResult<sgx_update_info_bit_t> {
+		todo!()
+	}
+
+	fn get_mrenclave_of_self(&self) -> SgxResult<sgx_measurement_t> {
+		Ok(sgx_measurement_t { m: self.mr_enclave })
+	}
+}
+
+impl EnclaveSidechainOCallApi for OnchainMock {
+	fn propose_sidechain_blocks<SB: Encode>(&self, _signed_blocks: Vec<SB>) -> SgxResult<()> {
+		Ok(())
+	}
+
+	fn store_sidechain_blocks<SB: Encode>(&self, _signed_blocks: Vec<SB>) -> SgxResult<()> {
+		Ok(())
+	}
+}
+
+// impl EnclaveOnChainOCallApi for OnchainMock {
```

Comment:
Yes, that's why I uncommented it again. The fact that we have this default implementation kind of restricts our mock. The fact that the `stf_executor` doesn't need that specific trait is fortunate - but it might not always work out that way. I will add a comment to this uncommented section 👍

Reply:
I chose the generic implementation because, we can't implement external traits for external types. Further, `EnclaveOnChainOcallApi` should not be used directly, only get storage verified, as we never want to read parentchain storage without verifying it.

So I think this constraint is a valid one.

Reply:
But we do use `EnclaveOnChainOCallApi` directly, that's the point.. The light-client uses it to send extrinsics to the parentchain.

Reply:
Oh, I see. This was not part of the design when I thought about the above. Hmm, maybe it makes sense to split the trait `EnclaveOnChainOcallApi` into `SendXtToParentchainOcallApi` and `GetParentchainStorageOcallApi`?

Reply:
We should then probably refactor this at some point, yes. As of now it's not a major problem - we have a working solution with mocking it for now.